config/src/lib.rs

```diff
@@ -158,6 +160,47 @@ impl ClientConfigBuilder {
     fn default_timeout(&self) -> Duration {
         Duration::from_secs(30)
     }
+
+    fn default_is_tls_enabled(&self) -> bool {
+        if let Some(controller_uri) = &self.controller_uri {
+            return match controller_uri.scheme() {
+                Ok(scheme) => scheme == TLS_SCHEME,
+                Err(_) => false,
+            };
+        }
+        false
+    }
+    /// validate the builder before returning it
+    ///
+    /// if both is_tls_enabled and controller_uri have been set
+    /// verify that the uri scheme (if present) matches the is_tls_enabled
+    /// value.
+    fn validate(&self) -> Result<(), String> {
+        if self.is_tls_enabled.is_none() || self.controller_uri.is_none() {
+            // at least one option has not been specified, cannot have a conflict
+            return Ok(());
+        }
+        let is_tls_enabled = self.is_tls_enabled.unwrap();
+        let scheme_result = self.controller_uri.as_ref().unwrap().scheme();
```

Comment:
Same here and below.

Reply:
`self.controller_uri.as_ref().unwrap()` is also guarded at line 179.  `scheme()` returns a String. I will also change this to re-check as requested.